From 5b80188e21cee4cee5f58b36dcdc11f77a5bc2e2 Mon Sep 17 00:00:00 2001
From: Hu Kejun <william.hu@rock-chips.com>
Date: Wed, 30 Sep 2020 16:58:47 +0800
Subject: [PATCH] media: rockchip: isp: set tmo bit in gain by tmo enable

normal mode also use tmo module now, so we should set tmo bit in gain by tmo enable

Signed-off-by: Hu Kejun <william.hu@rock-chips.com>
Change-Id: I35395cf20c53ab92645b419a5d162a23f65c5274
---
 drivers/media/platform/rockchip/isp/isp_params_v2x.c | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/drivers/media/platform/rockchip/isp/isp_params_v2x.c b/drivers/media/platform/rockchip/isp/isp_params_v2x.c
index 8cfbddbdf0ae..afaef878f55a 100644
--- a/drivers/media/platform/rockchip/isp/isp_params_v2x.c
+++ b/drivers/media/platform/rockchip/isp/isp_params_v2x.c
@@ -3433,23 +3433,20 @@ isp_gain_config(struct rkisp_isp_params_vdev *params_vdev,
 	struct rkisp_isp_params_val_v2x *priv_val =
 		(struct rkisp_isp_params_val_v2x *)params_vdev->priv_val;
 	u32 value, i, gain_wsize;
-	u8 tmo_en, mge_en;
+	u8 mge_en;
 
 	if (dev->csi_dev.rd_mode != HDR_NORMAL &&
-	    dev->csi_dev.rd_mode != HDR_RDBK_FRAME1) {
-		tmo_en = 1;
+	    dev->csi_dev.rd_mode != HDR_RDBK_FRAME1)
 		mge_en = 1;
-	} else {
-		tmo_en = 0;
+	else
 		mge_en = 0;
-	}
 
 	gain_wsize = rkisp_ioread32(params_vdev, MI_GAIN_WR_SIZE);
 	gain_wsize &= 0x0FFFFFF0;
 	if (gain_wsize)
 		value = (priv_val->dhaz_en & 0x01) << 16 |
 			(priv_val->wdr_en & 0x01) << 12 |
-			(tmo_en & 0x01) << 8 |
+			(priv_val->tmo_en & 0x01) << 8 |
 			(priv_val->lsc_en & 0x01) << 4 |
 			(mge_en & 0x01);
 	else
-- 
2.35.3

